hail/src/main/scala/is/hail/expr/ir/functions/GenotypeFunctions.scala

```diff
@@ -39,17 +39,17 @@ object GenotypeFunctions extends RegistryFunctions {
         m2 - m)
     }
 
-    registerCode[Long]("dosage", TArray(tv("N", "float64")), TFloat64,  (pt: PType) => PFloat64()) { case (r, rt, (gpPType, gpOff)) =>
-      val gpPArray = coerce[PArray](gpPType)
+    registerPCode("dosage", TArray(tv("N", "float64")), TFloat64,  (pt: PType) => PFloat64()
+    ) { case (r, rt, gp: PIndexableCode) =>
+      val code = EmitCodeBuilder.scopedCode[Double](r.mb) { cb =>
+        val gpv: PIndexableValue = gp.memoize(cb, "dosage_gp")
+        cb.ifx(gpv.loadLength().cne(3),
+          Code._fatal[Unit](const("length of gp array must be 3, got ").concat(gpv.loadLength().toS)))
 
-      Code.memoize(gpOff, "dosage_gp") { gp =>
-        Code.memoize(gpPArray.loadLength(gp), "dosage_len") { len =>
-        len.cne(3).mux(
-          Code._fatal[Double](const("length of gp array must be 3, got ").concat(len.toS)),
-          Region.loadDouble(gpPArray.elementOffset(gp, 3, 1)) +
-            Region.loadDouble(gpPArray.elementOffset(gp, 3, 2)) * 2.0)
-        }
+        gpv.loadElement(cb, 1).pc.tcode[Double] +
+          gpv.loadElement(cb, 2).pc.tcode[Double] * const(2.0)
```

Comment:
The issue is not defining the labels that are created when we `loadElement`. There are a few possible solutions. One is to make an unsafe version of loadElement that bypasses missing checks. Another is to just define the labels. Like so:
```suggestion
        val iec1 = gpv.loadElement(cb, 1)
        cb.define(iec1.Lmissing)
        cb.define(iec1.Lpresent)
        val iec2 = gpv.loadElement(cb, 2)
        cb.define(iec2.Lmissing)
        cb.define(iec2.Lpresent)

        iec1.pc.tcode[Double] + iec2.pc.tcode[Double] * 2.0
```

Reply:
the dosage code is incorrect -- it does need to check for missing! I assigned this task to Alex in Asana on Friday, but you can feel free to fix that here too (just reassign to yourself).

Just make sure you add a test!

Reply:
I'll take care of it then.